tests/rptest/utils/node_operations.py

```diff
@@ -79,6 +81,53 @@ def add(idx):
                 yield NodeOperation(op, idx, random.choice([True, False]))
 
 
+def verify_offset_translator_state_consistent(redpanda: RedpandaService):
+    logger = redpanda.logger
+    last_delta_pattern = re.compile('^\\{.*, last delta: (?P<delta>\\d+)\\}$')
+    admin = Admin(redpanda)
+    partitions = defaultdict(dict)
+
+    for n in redpanda.started_nodes():
+        node_id = redpanda.node_id(n)
+        all_partitions = admin.get_partitions(node=n)
+
+        def _state_consistent(ns, topic, partition):
+
+            state = admin.get_partition_state(ns, topic, partition, node=n)
+            dirty_offset = state['replicas'][0]['dirty_offset']
+            if all(r['dirty_offset'] == dirty_offset
+                   for r in state['replicas']):
+                return True, state
+            return False, None
+
+        for p in all_partitions:
+            namespace = p['ns']
+            topic = p['topic']
+            partition = p['partition_id']
+            state = wait_until_result(
+                lambda: _state_consistent(namespace, topic, partition),
+                timeout_sec=180,
+                backoff_sec=1,
+                err_msg="Error waiting for offsets to be consistent")
+
+            partitions[f"{namespace}/{topic}/{partition}"][node_id] = state
+
+    for p, node_states in partitions.items():
+        for n_id, state in node_states.items():
+            logger.debug(f"debug state of {p} replica on node {n_id}: {state}")
```

Comment:
nit: what's the point of the word `debug` in the message?

Reply:
the state is coming from `/v1/debug` REST API path